### ISSUE TITLE
stage2: builtin function used in block less if crashes compiler

### ISSUE THREAD

**Issue by FireFox317** [ziglang/zig#8458]
## Code
```zig
export fn _start() noreturn {
    var x: i32 = 1;
    if (x == 1) @breakpoint();
    while (true) {}
}
```
## ZIR
```zig
%9 = block({
    %5 = load(%2) node_offset:7:9
    %6 = cmp_eq(%5, @Ref.one) node_offset:7:11
    %7 = as_node(@Ref.bool_type, %6) node_offset:7:11
    %8 = condbr(%7, {
      %10 = breakpoint() node_offset:7:17
      %11 = break(%9, %10)
    }, {
      %12 = break(%9, @Ref.void_value)
    }) node_offset:7:5
  }) node_offset:7:5
  %13 = ensure_result_used(%9) node_offset:7:5
```
It can be clearly seen that a break is emited that referes to the `breakpoint` instruction, however the `breakpoint` instruction does not return a value. Ran into this when implementing `if (builtin.link_libc) @export(main, "main")`.
## Backtrace
```
Segmentation fault at address 0x0
/home/timon/dev/zig/zig-source/src/type.zig:296:35: 0x3aa23e in type.Type.eql (zig)
        if (a.tag_if_small_enough == b.tag_if_small_enough)
                                  ^
/home/timon/dev/zig/zig-source/src/Sema.zig:5196:29: 0x56d511 in Sema.resolvePeerTypes (zig)
        if (candidate.ty.eql(chosen.ty))
                            ^
/home/timon/dev/zig/zig-source/src/Sema.zig:1277:50: 0x56f7f0 in Sema.analyzeBlockBody (zig)
    const resolved_ty = try sema.resolvePeerTypes(parent_block, src, merges.results.items);
                                                 ^
/home/timon/dev/zig/zig-source/src/Sema.zig:1234:33: 0x4c12d3 in Sema.zirBlock (zig)
    return sema.analyzeBlockBody(parent_block, src, &child_block, merges);
                                ^
/home/timon/dev/zig/zig-source/src/Sema.zig:153:40: 0x4b81c1 in Sema.analyzeBody (zig)
            .block => try sema.zirBlock(block, inst),
                                       ^
/home/timon/dev/zig/zig-source/src/Sema.zig:69:28: 0x4b60f0 in Sema.root (zig)
    return sema.analyzeBody(root_block, root_body);
                           ^
/home/timon/dev/zig/zig-source/src/Module.zig:3658:22: 0x4258d6 in Module.analyzeFnBody (zig)
    _ = try sema.root(&inner_block);
                     ^
/home/timon/dev/zig/zig-source/src/Compilation.zig:1597:56: 0x3fb1e5 in Compilation.performAllTheWork (zig)
                        .queued => module.analyzeFnBody(decl, func) catch |err| switch (err) {
                                                       ^
/home/timon/dev/zig/zig-source/src/Compilation.zig:1388:31: 0x3f6842 in Compilation.update (zig)
    try self.performAllTheWork();
                              ^
/home/timon/dev/zig/zig-source/src/main.zig:2099:20: 0x387252 in updateModule (zig)
    try comp.update();
                   ^
/home/timon/dev/zig/zig-source/src/main.zig:1940:17: 0x32634d in buildOutputType (zig)
    updateModule(gpa, comp, hook) catch |err| switch (err) {
                ^
/home/timon/dev/zig/zig-source/src/main.zig:170:31: 0x310bb0 in mainArgs (zig)
        return buildOutputType(gpa, arena, args, .{ .build = .Exe });
                              ^
/home/timon/dev/zig/zig-source/src/main.zig:125:20: 0x308b8f in main (zig)
    return mainArgs(gpa, arena, args);
                   ^
/home/timon/dev/zig/zig-source/lib/std/start.zig:345:37: 0x2e0374 in std.start.posixCallMainAndExit (zig)
            const result = root.main() catch |err| {
                                    ^
/home/timon/dev/zig/zig-source/lib/std/start.zig:163:5: 0x2e0212 in std.start._start (zig)
    @call(.{ .modifier = .never_inline }, posixCallMainAndExit, .{});
```

**Comment by g-w1** [comment]
For some more debugging info: 
```
Thread 1 "zig" received signal SIGSEGV, Segmentation fault.
0x0000000000391b4e in type.Type.eql (a=..., 
    b=<error reading variable: Cannot access memory at address 0xaaaaaaaaaaaaaab2>) at type.zig:296
296             if (a.tag_if_small_enough == b.tag_if_small_enough)
(gdb) p a
$1 = {tag_if_small_enough = 27, ptr_otherwise = 0x1b}
(gdb) p b
Cannot access memory at address 0xaaaaaaaaaaaaaab2
(gdb) 
```
It seems like b is undefined in this case.

**Comment by FireFox317** [comment]
@g-w1 Yeah that makes sense, because `%11` should have never been emitted (in AstGen) and instead AstGen should have emitted a `break(%9, @Ref.void_value)` (similar to the else branch in the condbr)

**Comment by andrewrk** [comment]
Fixed in bcc371618fd71185e71a889c09eb68733ca66842
